python_modules/libraries/dagster-dg/dagster_dg/cli/check.py

```diff
@@ -72,12 +72,22 @@ def check_yaml_command(
     **global_options: object,
 ) -> None:
     """Check component.yaml files against their schemas, showing validation errors."""
-    resolved_paths = [Path(path).absolute() for path in paths]
-    top_level_component_validator = Draft202012Validator(schema=COMPONENT_FILE_SCHEMA)
-
     cli_config = normalize_cli_config(global_options, click.get_current_context())
     dg_context = DgContext.for_project_environment(Path.cwd(), cli_config)
 
+    if not check_yaml(dg_context, paths):
+        click.get_current_context().exit(1)
+    else:
+        click.echo("All components validated successfully.")
+
+
+def check_yaml(
```

Comment:
Now that this is logic is called from other commands, can we move it out of `dagster_dg.cli.check` into just `dagster_dg.check` (like `dagster_dg.scaffold`)? Helps keep the CLI module a clean interface layer.